src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameLexer.cs

```diff
@@ -288,6 +288,65 @@ public Result<StackFrameToken> TryScanPath()
             return CreateToken(StackFrameKind.NumberToken, GetSubSequenceToCurrentPos(start));
         }
 
+        /// <summary>
+        /// Scans a form similar to g__, where g is a GeneratedNameKind (a single character)
+        /// and identifier is valid identifier characters as with <see cref="TryScanIdentifier()"/>
+        /// </summary>
+        public Result<StackFrameToken> TryScanRequiredGeneratedNameSeparator()
+        {
+            var start = Position;
+            if (IsCharacter(CurrentChar))
```

Comment:
so it's ok that it's not necessary `g`?

Reply:
In this case I wanted the lexer to know that generated name separators are just `{letter}__` , and the parser determines whether the letter accounts for something that is supported.